src/Raven.Server/Documents/Handlers/BatchRequestParser.cs

```diff
@@ -315,7 +318,7 @@ private async Task<CommandData> MoveNextUnlikely(JsonOperationContext ctx)
                 if (_state.CurrentTokenType == JsonParserToken.EndArray)
                     return new CommandData { Type = CommandType.None };
 
-                return await ReadSingleCommand(ctx, _stream, _state, _parser, _buffer, _token);
+                return await ReadSingleCommand(ctx, _stream, _state, _parser, _buffer, modifier, _token);
```

Comment:
can we not pass a null here for most cases?

Reply:
I wanted to be on the safe side with this change. 
But we probably later, we can do a better reuse for the modifier.
Or I'm missing the point?

Reply:
The idea is that you are changing a lot of places, for single usecase.
Better to reduce the number of changes.